src/utils.ts

```diff
@@ -6,3 +15,61 @@ export function orderObject(unordered) {
       return obj;
     }, {});
 }
+
+export function createBoardMapOld(): BoardMap<{
+  captured: number;
+  captures: number;
+  revengeKills: number;
+}> {
+  /**
+   * i.e.
+   * a1: {
+   *   k: {
+   *    captured: 0,
+   *    captures: 0,
+   *   },
+   * },
+   */
+  const squareCaptureInfo: any = {}; // make typing simpler
+
+  for (const square of ALL_SQUARES) {
+    squareCaptureInfo[square] = {};
+    for (const piece of ALL_UNAMBIGUOUS_PIECE_SYMBOLS) {
+      squareCaptureInfo[square][piece] = {
+        captured: 0,
+        captures: 0,
+        revengeKills: 0,
+      };
+    }
+  }
+
+  return squareCaptureInfo;
+}
+
+/**
+ * A utitily function to create an object with unambiguous piece symbols as keys
+ */
+export function createUAPMap<T extends Object>(object: T): UAPMap<T> {
+  const map = {};
+  for (const uap of ALL_UNAMBIGUOUS_PIECE_SYMBOLS) {
+    map[uap] = { ...object };
+  }
+  return map as { [key in UASymbol]: T };
+}
+
+export function createBoardMap<T extends Object>(object: T): BoardMap<T> {
```

Comment:
in what contexts do you imagine these being used?

Reply:
I'm using it rn

Reply:
when we need a map of squares --> UAS --> some tracking counts